runtime/lua/vim/lsp/codelens.lua

```diff
@@ -279,25 +280,36 @@ function M.on_codelens(err, result, ctx, _)
   end)
 end
 
---- Refresh the codelens for the current buffer
+--- @class vim.lsp.codelens.RefreshOptions
+--- @field bufnr integer? filter by buffer. All buffers if nil, 0 for current buffer
+
+--- Refresh the lenses.
 ---
 --- It is recommended to trigger this using an autocmd or via keymap.
 ---
 --- Example:
 ---
 --- ```vim
---- autocmd BufEnter,CursorHold,InsertLeave <buffer> lua vim.lsp.codelens.refresh()
+--- autocmd BufEnter,CursorHold,InsertLeave <buffer> lua vim.lsp.codelens.refresh({ bufnr = 0 })
 --- ```
-function M.refresh()
+---
+--- @param opts? vim.lsp.codelens.RefreshOptions Table with the following fields:
+---  - `bufnr` (integer|nil): filter by buffer. All buffers if nil, 0 for current buffer
+function M.refresh(opts)
+  opts = opts or {}
+  local bufnr = opts.bufnr and resolve_bufnr(opts.bufnr)
+  local buffers = bufnr and { bufnr }
+    or vim.tbl_filter(api.nvim_buf_is_loaded, api.nvim_list_bufs())
   local params = {
     textDocument = util.make_text_document_params(),
   }
-  local bufnr = api.nvim_get_current_buf()
-  if active_refreshes[bufnr] then
-    return
+
+  for _, buf in ipairs(buffers) do
+    if not active_refreshes[buf] then
+      active_refreshes[buf] = true
+      vim.lsp.buf_request(buf, ms.textDocument_codeLens, params, M.on_codelens)
```

Comment:
Does the `util._refresh()` function save any code here? https://github.com/neovim/neovim/blob/5e5b004da44c7075ed1e20ae9d05ab09c6f2ac58/runtime/lua/vim/lsp/inlay_hint.lua#L251

Reply:
I don't think so, since `_refresh` treats `opts.bufnr` differently (here it is interpreted as all buffers, while `_refresh` treats it as 0) and it also includes `range` information in the parameters which we don't need here.

Of course we could modify it to be a common utility for inlay hints and codelens, but we want to burn `vim.lsp.util` anyway and I think keeping both implementations separate for now feels safer.

Reply:
Fair enough. Still, since the parameters have different interpretations, I think that the gain of a common function would be lost while handling the different routes. I believe that keeping the code separate is okay for now.

Reply:
I'm happy to refactor/move stuff around to reduce common code though :)

Reply:
Resolving this for now. The differences in the `refresh` functions are significant enough for a common utility not being worth it.